src/eventEmitter/EventEmitter.ts

```diff
@@ -0,0 +1,23 @@
+import { EventName, EventListener, Token } from "../types";
+
+export class EventEmitter {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  private events: { [key: string]: Function[] } = {};
+
+  on(event: EventName, listener: EventListener) {
+    if (!this.events[event]) {
+      this.events[event] = [];
+    }
+    this.events[event].push(listener);
+  }
+
+  off(event: EventName, listener: EventListener) {
+    if (!this.events[event]) return;
+    this.events[event] = this.events[event].filter((l) => l !== listener);
+  }
+
+  emit(event: EventName, payload: Token | Error) {
```

Comment:
is the payload type right here? or can it just be `unknown`?

Reply:
``` this._tokenEvents.emit("tokenRefreshFailed", error as Error);```
```this._tokenEvents.emit("tokenRefreshed", this.token);```

there are only two instance of emit. where payload is either Token or Error